R/combine_checkboxes.R

```diff
@@ -40,87 +41,84 @@
 combine_checkboxes <- function(supertbl,
                                tbl,
                                cols,
-                               values_to,
+                               names_prefix = "",
+                               names_suffix = NULL,
+                               names_sep = "_",
                                multi_value_label = "Multiple",
                                values_fill = NA,
                                raw_or_label = "label",
                                keep = TRUE) {
   # Check args ---
   check_arg_is_supertbl(supertbl, req_cols = c("redcap_data", "redcap_metadata"))
+  check_arg_is_character(names_prefix, len = 1)
+  check_arg_is_character(names_suffix, len = 1, null.ok = TRUE)
+  check_arg_is_character(names_sep, len = 1, any.missing = TRUE)
   check_arg_is_character(tbl, len = 1, any.missing = FALSE)
-  check_arg_is_character(values_to, len = 1, any.missing = FALSE)
   check_arg_is_character(multi_value_label, len = 1, any.missing = TRUE)
   check_arg_is_character(values_fill, len = 1, any.missing = TRUE)
   check_arg_choices(raw_or_label, choices = c("label", "raw"))
   check_arg_is_logical(keep, len = 1, any.missing = FALSE)
 
-  # Save user cols to enquosure
-  cols_exp <- enquo(cols)
-
   # Extract tbl from supertbl
   data_tbl <- supertbl %>%
     extract_tibble(tbl)
 
-  # Get field names from cols_exp, check that fields exist
-  field_names <- names(eval_select(cols_exp, data = data_tbl))
-  check_fields_exist(fields = field_names, expr = cols_exp)
+  # Save user cols to quo
+  cols_exp <- enquo(cols)
 
-  # Define values_to as the count of TRUEs/1s for the given checkbox field
-  # Assign TRUE if multiple selections made, and FALSE if one or zero made
-  data_tbl_mod <- data_tbl %>%
-    mutate(
-      !!values_to := case_when(
-        rowSums(select(., eval_tidy(cols_exp))) > 1 ~ TRUE,
-        TRUE ~ FALSE
-      )
-    )
+  # Evaluate the cols expression to get the selected column names
+  selected_cols <- names(eval_select(cols_exp, data = data_tbl))
+  check_fields_exist(fields = selected_cols, expr = cols_exp) # Check supplied fields exist
 
   # Get metadata reference table, check that chosen fields are checkboxes
-  metadata <- get_metadata_ref(data_tbl_mod, supertbl, tbl, field_names)
+  metadata_tbl <- supertbl$redcap_metadata[supertbl$redcap_form_name == tbl][[1]]
+  metadata_spec <- get_metadata_spec(metadata_tbl, selected_cols, names_prefix, names_suffix, names_sep)
```

Comment:
If we could get `names_glue` to work I would suggest not passing `names_prefix, names_suffix, names_sep` here and refactoring to have a function that converts those into an appropriate `names_glue` values